pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java

```diff
@@ -429,6 +443,12 @@ public SelectionResult select(BrokerRequest brokerRequest, List<String> segments
     }
   }
 
+  protected boolean isUseFixedReplica(Map<String, String> queryOptions) {
+    return Boolean.parseBoolean(
+        queryOptions.getOrDefault(CommonConstants.Broker.Request.QueryOptionKey.USE_FIXED_REPLICA,
```

Comment:
(minor) Suggest moving query option handling into `QueryOptionUtils`, and avoid per query string conversion:
```
  Boolean useFixedReplicaFromQueryOption = QueryOptionUtils.getUseFixedReplica(queryOptions);
  return useFixedReplicaFromQueryOption != null ? useFixedReplicaFromQueryOption : _useFixedReplica;
```

Reply:
Makes sense. Ack